include/dsn/dist/distributed_lock_service.h

```diff
@@ -26,19 +26,38 @@
 # pragma once
 
 # include <dsn/service_api_cpp.h>
+# include <string>
+# include <functional>
 
 namespace dsn
 {
     namespace dist
     {
-        // interface
         class distributed_lock_service
         {
         public:
-            // void create_lock(..callback.);
-            // void lock(..callback.);
-            // void unlock(..callback.);
-            // void destroy_lock(..callback.);
+            typedef void* lock_handle;
+
+            virtual lock_handle create_lock(const std::string& lock_name) = 0;
```

Comment:
A "create_lock" call only alloates contexts for the distribute_lock implemetation, so I think a callback may not be necessary